src/main/java/org/moditect/kcctl/service/compaction/KafkaDatabaseHistoryCompaction.java

```diff
@@ -0,0 +1,252 @@
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Collections;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.function.Consumer;
+
+public class KafkaDatabaseHistoryCompaction extends AbstractDatabaseHistoryCompaction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaDatabaseHistoryCompaction.class);
+
+    private static final String CONSUMER_PREFIX = CONFIGURATION_FIELD_PREFIX_STRING + "consumer.";
+    private static final String PRODUCER_PREFIX = CONFIGURATION_FIELD_PREFIX_STRING + "producer.";
+
+    public static final Field BOOTSTRAP_SERVERS = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "kafka.bootstrap.servers")
+            .withDisplayName("Kafka broker addresses")
+            .withType(ConfigDef.Type.STRING)
+            .withWidth(ConfigDef.Width.LONG)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDescription("A list of host/port pairs that the connector will use for establishing the initial "
+                    + "connection to the Kafka cluster for retrieving database schema history previously stored "
+                    + "by the connector. This should point to the same Kafka cluster used by the Kafka Connect "
+                    + "process.");
+
+    public static final Field RECOVERY_POLL_INTERVAL_MS = Field.create(CONFIGURATION_FIELD_PREFIX_STRING
+            + "kafka.recovery.poll.interval.ms")
+            .withDisplayName("Poll interval during database history recovery (ms)")
+            .withType(ConfigDef.Type.INT)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDescription("The number of milliseconds to wait while polling for persisted data during recovery.")
+            .withDefault(100)
+            .withValidation(Field::isNonNegativeInteger);
+
+    public static final Field RECOVERY_POLL_ATTEMPTS = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "kafka.recovery.attempts")
+            .withDisplayName("Max attempts to recovery database history")
+            .withType(ConfigDef.Type.INT)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW)
+            .withDescription("The number of attempts in a row that no data are returned from Kafka before recover completes. "
+                    + "The maximum amount of time to wait after receiving no data is (recovery.attempts) x (recovery.poll.interval.ms).")
+            .withDefault(100)
+            .withValidation(Field::isInteger);
+
+
+    /**
+     * The one and only partition of the history topic.
+     */
+    private static final Integer PARTITION = 0;
+    private static final int UNLIMITED_VALUE = -1;
+
+    private final DocumentReader reader = DocumentReader.defaultReader();
+    private final String topicName;
```

Comment:
What's topic name? The new topic? I'd expect this to be an option of the "compact" command?